org.adempiere.base/src/org/compiere/model/MAllocationHdr.java

```diff
@@ -327,17 +327,37 @@ protected boolean beforeDelete ()
 
 		//	Unlink
 		getLines(true);
-		if (!updateBP(true)) 
+		String prevStatus = getDocStatus();
+		setDocStatus(STATUS_Reversed);
+		if (!updateBP())
 			return false;
+		setDocStatus(prevStatus);

```

Comment:
this one is hard to understand - why this docstatus stuff is needed in beforeDelete ?